quickwit-metastore/src/checkpoint.rs

```diff
@@ -46,6 +46,20 @@ impl From<&str> for PartitionId {
     }
 }
 
+impl From<u32> for PartitionId {
+    fn from(partition_id: u32) -> Self {
+        let partition_id_str = format!("{:0>20}", partition_id);
```

Comment:
Let's not have this.
People can cast.

Also if such converters were needed, they should all call into the same converter.

Reply:
Ah, yes of course, showing my rust newbie-ness.